spot_ros2_control/hardware/src/spot_hardware_interface.cpp

```diff
@@ -373,6 +411,114 @@ void SpotHardware::stop_state_stream() {
   state_stream_started_ = false;
 }
 
+bool SpotHardware::start_command_stream() {
+  if (command_stream_started_) {
+    RCLCPP_INFO(rclcpp::get_logger("SpotHardware"), "Command stream has already been started!");
+    return true;
+  }
+  // Start command streaming
+  auto robot_command_stream_client_resp = robot_->EnsureServiceClient<::bosdyn::client::RobotCommandClient>();
+  if (!robot_command_stream_client_resp) {
+    RCLCPP_ERROR(rclcpp::get_logger("SpotHardware"), "Could not create robot command client");
+    return false;
+  }
+  command_client_ = robot_command_stream_client_resp.response;
+
+  auto endpoint_result = robot_->StartTimeSyncAndGetEndpoint();
+  if (!endpoint_result) {
+    RCLCPP_ERROR(rclcpp::get_logger("SpotHardware"), "Could not get timesync endpoint");
+    return false;
+  }
+
+  command_client_->AddTimeSyncEndpoint(endpoint_result.response);
+
+  RCLCPP_INFO(rclcpp::get_logger("SpotHardware"), "Robot Command Client successfully created!");
+
+  bosdyn::api::RobotCommand joint_command = ::bosdyn::client::JointCommand();
+  auto joint_res = command_client_->RobotCommand(joint_command);
+  if (!joint_res.status) {
+    RCLCPP_ERROR(rclcpp::get_logger("SpotHardware"), "Failed to activate joint control mode");
+    RCLCPP_ERROR(rclcpp::get_logger("SpotHardware"), "Message: %s", joint_res.status.DebugString().c_str());
+    return false;
+  }
+
+  auto robot_command_stream_resp = robot_->EnsureServiceClient<::bosdyn::client::RobotCommandStreamingClient>();
+  if (!robot_command_stream_resp) {
+    RCLCPP_ERROR(rclcpp::get_logger("SpotHardware"), "Could not create robot command streaming client");
+    return false;
+  }
+  command_stream_service_ = robot_command_stream_resp.response;
+
+  RCLCPP_INFO(rclcpp::get_logger("SpotHardware"), "Robot Command Streaming Client successfully created!");
+
+  // Fill in the parts of the joint streaming command request that are constant.
+  auto* joint_cmd = joint_request_.mutable_joint_command();
+  // Gain values https://github.com/boston-dynamics/spot-cpp-sdk/blob/master/cpp/examples/joint_control/constants.hpp
+  // NOTE: these should be different depending on number of joints the robot has (arm or none)
+  // Right now this is just temporary to see if we can get the commands accepted by robot
+  // This should be handled via a parameter in the future.
+  std::vector<float> kp;
+  std::vector<float> kd;
+  if (njoints_ == 19) {
+    kp = {624, 936, 286, 624, 936, 286, 624, 936, 286, 624, 936, 286, 1020, 255, 204, 102, 102, 102, 16.0};
+    kd = {5.20, 5.20, 2.04, 5.20, 5.20, 2.04, 5.20, 5.20, 2.04, 5.20,
+          5.20, 2.04, 10.2, 15.3, 10.2, 2.04, 2.04, 2.04, 0.32};
+  } else if (njoints_ == 12) {
+    kp = {624, 936, 286, 624, 936, 286, 624, 936, 286, 624, 936, 286};
+    kd = {5.20, 5.20, 2.04, 5.20, 5.20, 2.04, 5.20, 5.20, 2.04, 5.20, 5.20, 2.04};
+  } else {
+    RCLCPP_ERROR(rclcpp::get_logger("SpotHardware"), "WRONG # OF JOINTS");
+    return false;
+  }
```

Comment:
(I know I added this code but) can we define these elsewhere? 
Maybe we can have a common header that has these default kp and kd gains, and also can contain the map from joint name -> joint index

Reply:
i agree lets move the gains and joint index map to a headerfile or config somewhere. We could potentially just throw it into the hardware interface header file but maybe thats a bad idea